File: app/assets/javascripts/accounts_app/models/line_item.js
AccountsApp.Models.LineItem = Backbone.DeepModel.extend({
  
  initialize: function() {

  },

  schema: {
    short_desc: { title: 'Title', type: 'Text' },
    long_desc: { title: 'Description', type: 'Text' },
    owner_id: { title: 'Owner',
      type: 'Select', 
      options: function(callback){
        callback(AccountsApp.users );
      }
    },
    eff_date: { title: 'Effective Date', type: 'Date' }
  },

  urlRoot: '/line_items',

  getOwner: function() {
    if ( ! this.owner ) 
    {
      var owner_id = this.get('owner_id');
      if (owner_id ) {
        owner = AccountsApp.users.get(owner_id);
        this.owner = owner;
      }
    }

    return this.owner;
  },

  toString: function() {
    return this.get('short_desc');
  }
 
});

File: app/models/line_item.rb
class LineItem < ActiveRecord::Base
  belongs_to :owner, :class_name => 'User'
  belongs_to :account

  has_many :claims

  def as_json( options={} ) 
    {
      id: self.id,
      owner_id:  self.owner_id ,
      short_desc: self.short_desc,
      eff_date: self.eff_date,
    }
  end

end
